Formula/l/llvm.rb

```diff
@@ -665,6 +670,11 @@ def caveats
         }
       EOS
 
+      system "echo", "'DEBUG missing libomp.so'"
+      system "find", HOMEBREW_PREFIX.to_s, "-name", "libomp*.so", "-print"
+      system "echo", env.LD_LIBRARY_PATH
+      system "find", "/usr", "-name", "libomp*.so", "-print"
+
       system bin/"clang", "-L#{lib}", "-fopenmp", "-nobuiltininc",
                              "-I#{lib}/clang/#{llvm_version_major}/include",
                              "omptest.c", "-o", "omptest"
```

Comment:
```suggestion
      system bin/"clang", "-L#{lib}", "-fopenmp", "-nobuiltininc",
                          "-I#{lib}/clang/#{llvm_version_major}/include",
                          "-Wl,-rpath,#{lib}/x86_64-unknown-linux-gnu",
                          "omptest.c", "-o", "omptest"
```

Reply:
Though should probably scope that flag to Linux only.

Reply:
Thanks, will try that. If it fixes the Linux job, I'll put an OS condition on it.

Reply:
I still haven't been able to get a Linux runner started to test this fix, but I added the OS/arch condition when I updated to rc4.

Reply:
Green on Linux with your fix https://github.com/Homebrew/homebrew-core/actions/runs/10733868906/job/29768247878?pr=181351